node/network/collator-protocol/src/validator_side/mod.rs

```diff
@@ -761,16 +754,31 @@ where
 			};
 
 			match peer_data.insert_advertisement(relay_parent, &state.view) {
-				Ok((collator_id, para_id)) => {
+				Ok((id, para_id)) => {
 					tracing::debug!(
 						target: LOG_TARGET,
 						peer_id = ?origin,
 						%para_id,
 						?relay_parent,
 						"Received advertise collation",
 					);
+					let (tx, rx) = oneshot::channel::<(
+						CandidateReceipt,
+						PoV,
+					)>();
+
+
+					let pending_collation = PendingCollation::new(
+						relay_parent,
+						&para_id,
+						&origin,
+					);
+					fetch_collation(ctx, state, pending_collation.clone(), tx).await;
 
-					notify_candidate_selection(ctx, collator_id, relay_parent, para_id).await;
+					let future = async move {
+						((id, pending_collation), rx.timeout(COLLATION_FETCH_TIMEOUT).await)
```

Comment:
We already have a timeout in the request-response system. That is 1 second currently. So, we don't really need this?

Reply:
Ah, you're right https://github.com/paritytech/polkadot/blob/a4dfdf160c527ebd739dc24bb5afa453f9293a98/node/network/protocol/src/request_response/mod.rs#L135

Yes, it can be removed.